doc/source/ray-core/patterns/unnecessary-ray-get.rst

```diff
@@ -0,0 +1,34 @@
+Anti-pattern: Calling ray.get unnecessarily harms performance
+=============================================================
+
+**TLDR:** Avoid calling :ref:`ray.get() <ray-get-ref>` unnecessarily for intermediate steps. Work with object references directly, and only call ``ray.get()`` at the end to get the final result.
+
+When ``ray.get()`` is called, objects must be transferred to the worker/node that calls ``ray.get()``. If you don't need to manipulate the object, you probably don't need to call ``ray.get()`` on it!
+
+Typically, it’s a best practice to wait as long as possible before calling ``ray.get()``, or even design your program to avoid having to call ``ray.get()`` at all.
```

Comment:
```suggestion
Typically, it’s best practice to wait as long as possible before calling ``ray.get()``, or even design your program to avoid having to call ``ray.get()`` at all.
```